### PR TITLE
Exit vespa cli with non-zero exit code on error

### PR DESCRIPTION
**Pull request by hakonhall**

### CODE REVIEW

File: client/go/cmd/vespa/main.go [hunk 1]
```diff
@@ -6,8 +6,11 @@ package main
 
 import (
 	"github.com/vespa-engine/vespa/client/go/cmd"
+	"os"
 )
 
 func main() {
-	cmd.Execute()
+	if cmd.Execute() != nil {
```
Comment (mpolden) [inline client/go/cmd/vespa/main.go:13]: Nit: `if err := ...; err != nil {` is a common idiom to indicate that the returned value is in fact an error.

**Review by mpolden — APPROVED** [review]

File: client/go/cmd/vespa/main.go (hunk 1 above)
Comment (mpolden) [reply client/go/cmd/vespa/main.go:13]: Too bad `cobra.Command{Run: ...}` does not return an error, because then we could pass all errors up to here.

Comment (hakonhall) [reply client/go/cmd/vespa/main.go]: Which kind of errors are masked by cobra.Command?

Comment (hakonhall) [reply client/go/cmd/vespa/main.go]: Fixed

**Pull request merged**